src/Chapter3.hs

```diff
@@ -374,6 +381,28 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Knight = MkKnight
+    { knightHealth :: Int
+    , knightAttack :: Int
+    , knightGold :: Int
+    } deriving (Show)
+
+data Monster = MkMonster
+    { monsterHealth :: Int
+    , monsterAttack :: Int
+    , monsterGold :: Int
+    } deriving (Show)
+
+fight :: Monster -> Knight -> Int
+fight monster knight =
+    let monsterHealth' = monsterHealth monster - knightAttack knight
+        knightHealth' = knightHealth knight - monsterAttack monster
+    in
+    if monsterHealth' > 0 then
+        if knightHealth' < 0 then -1 else knightGold knight
+    else
+        knightGold knight + monsterGold monster
```

Comment:
That's a correction solution :+1: 

But I feel like it would look more elegant with guards instead of nested if-then-else expressions :slightly_smiling_face: